bloxapp/ssv-dkg
# Deposit validation

We need to double check `deposit_file.json` before it gets submitted to `mainnet`.

* added partial `launchpad` checks as we want to validate BLS signature using JS library before submitting the deposit.
* copy `launchpad` validation logic to `golang`

To run:
1. Build docker image 
```sh
make docker-build-deposit-verify
```
2. Run docker container with 2 params: `DEPOSIT_FILE_PATH=[path to deposit_file.json]` and `NETWORK_ENV_PATH=[path to network env (e.g. .env.holesky)]`

```sh
make DEPOSIT_FILE_PATH=/home/pk/go/src/github.com/bloxapp/ssv-dkg/examples/deposit_data.json NETWORK_ENV_PATH=/home/pk/go/src/github.com/bloxapp/ssv-dkg/utils/deposit_verify/.env.holesky docker-deposit-verify
```
